File: trunk/tm/src/tm/portableDisplays/StateInterface.java
package tm.portableDisplays;

public interface StateInterface {

	public String getExpression();
	public CodeLine getSelectedCodeLine(SuperTMFile tmFile, boolean allowGaps, int i);
	public int getNumSelectedCodeLines(SuperTMFile tmFile, boolean allowGaps);
	public SuperSourceCoords getCodeFocus() ;
	public void setSelection(SelectionInterface selection) ;
	public SelectionInterface getSelection() ;
}

File: trunk/JSTM/src/StateInterface.ts
module jstm {
    
    export interface StateInterface {
        
        getExpression : () => string ;

        getNumConsoleLines : () => number ;

        getConsoleLine : (i : number) => string ;

        getNumSelectedCodeLines : (tmf : TMFileI, allowGaps : boolean ) => number ;

        getSelectedCodeLine : (tmf : TMFileI, allowGaps : boolean, i : number) => CodeLineI ;

        getSelection : () => SelectionInterface ;
    }
}
